lxd/db/instances.go

```diff
@@ -872,6 +872,29 @@ func (c *ClusterTx) GetInstanceSnapshotsWithName(ctx context.Context, project st
 	return instances, nil
 }
 
+// GetLocalInstanceWithVsockID returns all available instances with the given config key and value.
+func (c *ClusterTx) GetLocalInstanceWithVsockID(vsockID int) (*cluster.Instance, error) {
+	q := `
+SELECT instances.id, projects.name AS project, instances.name, nodes.name AS node, instances.type, instances.architecture, instances.ephemeral, instances.creation_date, instances.stateful, instances.last_use_date, coalesce(instances.description, ''), instances.expiry_date
+  FROM instances JOIN projects ON instances.project_id = projects.id JOIN nodes ON instances.node_id = nodes.id JOIN instances_config ON instances.id = instances_config.instance_id
+  WHERE instances.node_id = ? AND instances.type = ? AND instances_config.key = "volatile.vsock_id" AND instances_config.value = ? ORDER BY projects.id, instances.name
```

Comment:
The `ORDER BY` isn't needed, and adding a `LIMIT 1` may help the query optimizer end the query quicker after finding the first row.